domain/info.go

```diff
@@ -144,6 +158,47 @@ func (is *InfoSyncer) RemoveServerInfo() {
 	}
 }
 
+// storeMinStartTS stores self server min start timestamp to etcd.
+func (is *InfoSyncer) storeMinStartTS(ctx context.Context) error {
+	if is.etcdCli == nil {
+		return nil
+	}
+	return util.PutKVToEtcd(ctx, is.etcdCli, keyOpDefaultRetryCnt, is.minStartTSPath,
+		strconv.FormatUint(is.minStartTS, 10),
+		clientv3.WithLease(is.session.Lease()))
```

Comment:
Does all TiDB servers updates the same key in etcd? If so, how do TiDB servers know whether there's another TiDB server who has a smaller `minStartTs`?

Reply:
Oh I got it. LGTM.